projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php

```diff
@@ -796,24 +965,25 @@ public function is_in_disallowed_list( $in_disallowed_list, $form = array() ) {
 	 * Populate an array with all values necessary to submit a NEW contact-form feedback to Akismet.
 	 * Note that this includes the current user_ip etc, so this should only be called when accepting a new item via $_POST
 	 *
-	 * @param array $form Contact form feedback array
-	 * @return array feedback array with additional data ready for submission to Akismet
+	 * @param array $form - contact form feedback array.
+	 *
+	 * @return array feedback array with additional data ready for submission to Akismet.
 	 */
-	function prepare_for_akismet( $form ) {
+	public function prepare_for_akismet( $form ) {
 		$form['comment_type'] = 'contact_form';
-		$form['user_ip']      = $_SERVER['REMOTE_ADDR'];
-		$form['user_agent']   = isset( $_SERVER['HTTP_USER_AGENT'] ) ? $_SERVER['HTTP_USER_AGENT'] : '';
-		$form['referrer']     = isset( $_SERVER['HTTP_REFERER'] ) ? $_SERVER['HTTP_REFERER'] : '';
+		$form['user_ip']      = isset( $_SERVER['REMOTE_ADDR'] ) ? sanitize_text_field( wp_unslash( $_SERVER['REMOTE_ADDR'] ) ) : '';
+		$form['user_agent']   = isset( $_SERVER['HTTP_USER_AGENT'] ) ? sanitize_text_field( wp_unslash( $_SERVER['HTTP_USER_AGENT'] ) ) : '';
+		$form['referrer']     = isset( $_SERVER['HTTP_REFERER'] ) ? sanitize_text_field( wp_unslash( $_SERVER['HTTP_REFERER'] ) ) : '';
```

Comment:
`sanitize_text_field` could easily mangle the URI by deleting percent-encoded characters.